src/gtk/clipbrd.cpp

```diff
@@ -595,8 +597,19 @@ void wxClipboard::Clear()
 
 bool wxClipboard::Flush()
 {
-    gtk_clipboard_store( gtk_clipboard_get( GTKGetClipboardAtom() ) );
-    return true;
+    // Only store the non-primary clipboard when flushing. The primary clipboard is a scratch-space
+    // formed using the currently selected text.
+    if ( !m_usePrimary )
+    {
+        GtkClipboard* clipboard = gtk_clipboard_get(GDK_SELECTION_CLIPBOARD);
```

Comment:
I'm a bit surprised this is not `g_clipboardAtom`. Should we use `GDK_SELECTION_CLIPBOARD` instead of it elsewhere too then?

Reply:
I don't think it matters which we use, since they both will point to the same actual `GdkAtom` I believe. We already have a mix of them though (the primary clipboard is implemented using `GDK_SELECTION_PRIMARY` everywhere). It does appear though that they have removed `gdk_atom_intern` in GTK4 (https://gitlab.gnome.org/GNOME/gtk/-/commit/54a62731916bbf724b53f0c9fc2bd1addb85802f#f49af20942c0779b40da5fc0b8af38b3ed84aacc), so the code will probably need to be updated at some point, but at that time it might be good to investigate the actual GTK clipboard API functions (https://developer.gnome.org/gtk3/stable/gtk3-Clipboards.html) instead of using the selection API functions.

Reply:
> both will point to the same actual `GdkAtom`

This was the part I wasn't completely sure about: `GDK_SELECTION_CLIPBOARD` is defined as `GDK_MAKE_ATOM(69)` while our `g_clipboardAtom` is `gdk_atom_intern("CLIPBOARD")`. I don't quite understand how do they manage to be the same, but if you confirm they're, this is fine. If there is any doubt, and if it also works (I have no idea how to test this), I'd rather use `g_clipboardAtom` for consistency.